In [84]:
import numpy as np
import pandas as pd
from scipy import stats

### Задание 1.

Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks):
zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110],
ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832].
Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy
Полученные значения должны быть равны.
Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков,
а затем с использованием функций из библиотек numpy и pandas.

#### Решение:

$$cov_{xy} = M(XY) - M(X) \cdot M(Y)$$

$$r_{XY} = \frac{cov_{XY}}{\sigma_{X}\sigma_{Y}}$$

In [2]:
zp = np.array([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])
ks = np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])

In [15]:
cov = np.mean(zp * ks) - np.mean(zp) * np.mean(ks)
cov

9157.839999999997

In [18]:
np.cov(zp,ks,ddof=0)

array([[ 3494.64,  9157.84],
       [ 9157.84, 30468.89]])

In [12]:
r = cov/(np.std(zp) * np.std(ks))
r

0.8874900920739158

In [ ]:
np.corrcoef(zp, ks)

array([[1.        , 0.88749009],
       [0.88749009, 1.        ]])

In [36]:
df = pd.DataFrame({'zp': zp, 'ks': ks})
df

zp   ks
0   35  401
1   45  574
2  190  874
3  200  919
4   40  459
5   70  739
6   54  653
7  150  902
8  120  746
9  110  832

In [39]:
corr_matrix = df.corr()
corr_matrix.loc['zp', 'ks']

0.8874900920739168

### Задание 2.

Измерены значения IQ выборки студентов,
обучающихся в местных технических вузах:
131, 125, 115, 122, 131, 115, 107, 99, 125, 111.
Известно, что в генеральной совокупности IQ распределен нормально.
Найдите доверительный интервал для математического ожидания с надежностью 0.95.

#### Решение:

Стандартное отклонение генеральной совокупности неизвестно, поэтому используем критерий Стьюдента.
$$T_{1,2}=\overline{X}\pm t_{\frac{\alpha}{2}} \frac{\sigma_н}{\sqrt{n}}$$

In [67]:
iq = np.array([131, 125, 115, 122, 131, 115, 107, 99, 125, 111])
X = np.mean(iq)

In [68]:
sigma_n = np.std(iq, ddof = 1)

In [74]:
dfm = len(iq)-1
alpha = 1-0.95

In [73]:
t = stats.t.ppf(1.0 - alpha/2, dfm)
t

2.2621571627409915

In [70]:
T1 = X - t*(sigma_n/np.sqrt(dfm))
T1

110.1480139537451

In [71]:
T2 = X + t*(sigma_n/np.sqrt(dfm))
T2

126.05198604625488

#### Ответ

$T_{1,2} = [110; 126]$

### Задание 3.

Известно, что рост футболистов в сборной распределен нормально
с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27,
среднее выборочное составляет 174.2. Найдите доверительный интервал для математического
ожидания с надежностью 0.95.

#### Решение:

Стандартное отклонение генеральной совокупности известно, поэтому используем Z-критерий.
$$T_{1,2}=\overline{X}\pm Z_{\frac{\alpha}{2}} \frac{\sigma}{\sqrt{n}}$$

In [77]:
sigma = np.sqrt(25)
n = 27
X = 174.2
alpha = 1-0.95

In [87]:
Z = stats.norm.ppf(alpha/2)
Z

-1.959963984540054

In [91]:
T1 = X + Z * (sigma/np.sqrt(27))
T1

172.3140237765397

In [92]:
T2 = X - Z * (sigma/np.sqrt(27))
T2

176.08597622346028

#### Ответ

$T_{1,2} = [172; 176]$